embasi/embedding.py

```diff
@@ -906,3 +919,173 @@ def run(self):
+
+        initial_calculator.parameters["aims_output"] = "rho_and_derivs_on_grid"
+        low_level_calculator.parameters['aims_output'] = "rho_and_derivs_on_grid"
+        high_level_calculator.parameters['aims_output'] = "rho_and_derivs_on_grid"
+
+        self.set_layer(atoms, "MU0", initial_calculator, 
+                       embed_mask, ghosts=2, no_scf=False,
+                       insert_embedding_region=False)
+
+        self.set_layer(atoms, "F2A1", low_level_calculator, 
+                       embed_mask, ghosts=2, no_scf=False,
+                       insert_embedding_region=False)
+
+        self.set_layer(atoms, "F1A2", high_level_calculator,
+                       embed_mask, ghosts=1, no_scf=False,
+                       insert_embedding_region=False)
+
+        self.rank = MPI.COMM_WORLD.Get_rank()
+        self.ntasks = MPI.COMM_WORLD.Get_size()
+
+    def get_embedding_pot(self, atomsembed, n_scf, ):
+        return 
+
+    def run(self):
+        """ Summary 
+        """
+        import numpy as np
+
+        root_print("Embedding calculation begun...")
+
+        start = time.time()
+
+        ediff = 1.0
+        etot_prev = 0.0
+        max_cycle = 50
+        n_cycle = 0
+
+        cwd = os.path.join(os.getcwd(),self.run_dir[2:])
+    
+        MU0_opot  = os.path.join(cwd, "MU0/output_potential_00000000")
+        F2A1_opot = os.path.join(cwd, "F2A1/output_potential_00000000")
+        F1A2_opot = os.path.join(cwd, "F1A2/output_potential_00000000")
+
+        F2A1_npot = os.path.join(cwd, "F2A1/output_potential_00000000_new")
+        F1A2_npot = os.path.join(cwd, "F1A2/output_potential_00000000_new")
+
+        MU0_oden  = os.path.join(cwd, "MU0/output_density_00000000")
+        F2A1_oden = os.path.join(cwd, "F2A1/output_density_00000000")
+        F1A2_oden = os.path.join(cwd, "F1A2/output_density_00000000")
+
+        F2A1_nden = os.path.join(cwd, "F2A1/output_density_00000000_new")
+        F1A2_nden = os.path.join(cwd, "F1A2/output_density_00000000_new")
+
+        # Outer SCF loop
+        fdet_file = open("fdet.txt", "w")
+        while np.abs(ediff) > 1.e-6 and n_cycle < max_cycle:
+            n_cycle += 1
+            print("SCF cycle: ", n_cycle)
+
+            if n_cycle == 1:
+                self.MU0.run()
+                try:
+                    tmp = os.path.join(self.run_dir, "F2A1")
+                    os.mkdir(tmp)
+                except FileExistsError:
+                    root_print("Directory F2A1 already exists")
+                try:
+                    tmp = os.path.join(self.run_dir, "F1A2")
+                    os.mkdir(tmp)
+                except FileExistsError:
+                    root_print("Directory F1A2 already exists")
+
+                shutil.copy(MU0_oden, F1A2_oden)
+                shutil.copy(MU0_opot, F1A2_opot)
+
+            # Cluster Calculation
+            self.F1A2.run()
```

Comment:
Maybe a small print statement here to let the user know which calculation is presently running - if the calculation is hanging on a particular job, it can allow some diagnosis of where the calculation is at.